libraries/Microsoft.Bot.Builder/MemoryStorage.cs

```diff
@@ -18,7 +18,7 @@ public class MemoryStorage : IStorage
     {
         private static readonly JsonSerializer StateJsonSerializer = new JsonSerializer()
         {
-            TypeNameHandling = TypeNameHandling.All,
+            TypeNameHandling = TypeNameHandling.None,
```

Comment:
The fact this works with all of the tests is awesome!  But, this change has the potential to inadvertently break customer unit tests.  We had to change our tests to support both.

Reply:
I agree with @EricDahlvang here, this change breaks the corebot sample. I think the default should be All (like it was) and we should add an option in the constructor so the developer can change the serialization settings to None if they need (like we do with ComsbosDbStorage and AzureBlobStorage). 
The ideal solution is to use None so we should still change our samples to use the new constructor and the code so things work but I wouldn't change the default behavior so we don't break existing bots. 
We should plan to default to None in our next major release.